src/jit/emitxarch.cpp

```diff
@@ -3956,21 +4053,44 @@ void emitter::emitIns_R_R_I(instruction ins, emitAttr attr, regNumber reg1, regN
 
     instrDesc* id = emitNewInstrSC(attr, ival);
 
-    // REX prefix
-    if (IsExtendedReg(reg1, attr) || IsExtendedReg(reg2, attr))
-    {
-        sz += emitGetRexPrefixSize(ins);
-    }
-
-    if ((ins == INS_pextrq || ins == INS_pinsrq) && !UseVEXEncoding())
-    {
-        sz += 1;
-    }
-
     id->idIns(ins);
     id->idInsFmt(IF_RRW_RRW_CNS);
     id->idReg1(reg1);
     id->idReg2(reg2);
+
+    code_t code = 0;
+
+    switch (ins)
```

Comment:
I grepped for all the callers to `emitIns_R_R_I` and believe I have them all covered. Most direct callers are either from simdcodegen or hwintrinsiccodegen. Most indirect callers go through `RV_RV_IV`, which itself has an assert on which instructions it should see.